goldenmaster/Plugins/TbEnum/Source/TbEnum/Private/Generated/OLink/TbEnumEnumInterfaceOLinkAdapter.cpp

```diff
@@ -0,0 +1,133 @@
+		BackendService->GetProp3ChangedDelegate().RemoveDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnProp3Changed);
+		BackendService->GetSig0SignalDelegate().RemoveDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig0);
+		BackendService->GetSig1SignalDelegate().RemoveDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig1);
+		BackendService->GetSig2SignalDelegate().RemoveDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig2);
+		BackendService->GetSig3SignalDelegate().RemoveDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig3);
+	}
+
+	// only set if interface is implemented
+	checkf(InService.GetInterface() != nullptr, TEXT("Cannot set backend service - interface HmiFrontDisplay is not fully implemented"));
+
+	// subscribe to new backend
+	BackendService = InService;
+	// connect property changed signals or simple events
+	BackendService->GetProp0ChangedDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnProp0Changed);
+	BackendService->GetProp1ChangedDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnProp1Changed);
+	BackendService->GetProp2ChangedDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnProp2Changed);
+	BackendService->GetProp3ChangedDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnProp3Changed);
+	BackendService->GetSig0SignalDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig0);
+	BackendService->GetSig1SignalDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig1);
+	BackendService->GetSig2SignalDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig2);
+	BackendService->GetSig3SignalDelegate().AddDynamic(this, &UTbEnumEnumInterfaceOLinkAdapter::OnSig3);
+
+	// update olink source with new backend
+	Source->setBackendService(InService);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnSig0(ETbEnumEnum0 Param0)
+{
+	Source->OnSig0(Param0);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnSig1(ETbEnumEnum1 Param1)
+{
+	Source->OnSig1(Param1);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnSig2(ETbEnumEnum2 Param2)
+{
+	Source->OnSig2(Param2);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnSig3(ETbEnumEnum3 Param3)
+{
+	Source->OnSig3(Param3);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnProp0Changed(ETbEnumEnum0 InProp0)
+{
+	Source->OnProp0Changed(InProp0);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnProp1Changed(ETbEnumEnum1 InProp1)
+{
+	Source->OnProp1Changed(InProp1);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnProp2Changed(ETbEnumEnum2 InProp2)
+{
+	Source->OnProp2Changed(InProp2);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::OnProp3Changed(ETbEnumEnum3 InProp3)
+{
+	Source->OnProp3Changed(InProp3);
+}
+
+void UTbEnumEnumInterfaceOLinkAdapter::setOLinkHost(TSoftObjectPtr<UUnrealOLinkHost> InHost)
+{
+	check(InHost);
+	Source->setOLinkHost(InHost.Get());
+
+	// register source to host
+	InHost->GetOLinkRegistry()->addSource(Source);
```

Comment:
good catch

Reply:
Should not be a problem atm since there can only be one host.
Anyway removing the source from the old registry now,
but should be revisited once we allow several hosts.